src/components/Layout/NavBar/NavBar.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <div class="topnav" @mouseleave="closeMenu">
+    <div id="menu_button" class="topnav-icon" @click="toggleMenu">
+      <i class="fa fa-navicon"></i>
+    </div>
+    <div :class="menuOpen ? 'open' : ''" id="fixed-menu">
+      <NavMenu />
+    </div>
+    <div class="topnav-logo"></div>
+    <div class="topnav-collapse-wrapper">
+      <div class="topnav-app-name">Community App</div>
+    </div>
+  </div>
+</template>
+
+<script>
+import NavMenu from './NavMenu.vue';
+
+export default {
+  name: 'NavBar',
+  data() {
+    return {
+      menuOpen: false,
+    };
+  },
+  methods: {
+    toggleMenu() {
+      this.menuOpen = !this.menuOpen;
+    },
+    closeMenu() {
+      this.menuOpen = false;
+    },
+  },
+  components: {
+    NavMenu,
+  },
+};
+</script>
+
+<style scoped>
```

Comment:
@ewenjo @spiray @rabbitwerks @AndCotOli Shouldn't we use separate files for CSS? Like a component with a separate folder with its CSS and stuff. What do you guys think?

Reply:
no, in keeping with Vue best practices component styles should live inside the single file component, so that everything is contained and scoped. 

anything being used across multiple components, like an input-base style would be moved to a globally accessed main.css file

Reply:
in this being the case, the css is too specific as it stands, we can re-write it to not be so over-redundant, since component styles are being scoped with the `scoped` attribute. I left a number of comments on the original PR way above, please check those out as a reference. Thank! :)

Reply:
Ah! Scoped. That's right. We can't use that privilege outside these files. Thanks for clarifying. :)